fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/services/functions/HierarchicalIdentifierSupplier.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api.services.functions;
+
+import static java.util.UUID.randomUUID;
+
+import java.util.stream.IntStream;
+import java.util.StringJoiner;
+
+/**
+ * Unique value minter that creates hierarchical IDs from a UUID
+ *
+ * @author awoods
+ * @author acoburn
+ */
+public interface HierarchicalIdentifierSupplier extends UniqueValueSupplier {
```

Comment:
Why is this new class `HierarchicalIdentifierSupplier` in the `org.fcrepo.kernel.api.services.functions` package?